### DISCUSSION TITLE
v1.7.0 release

### DISCUSSION THREAD

**Discussion by ctlcltd** [ctlcltd/e2-sat-editor#29] [Announcements]
**[DEPRECATED VERSION]**

## Changelog
- New utils: autofix, remove duplicates, fix/remove parameters and flags, sort
- Improved lists compatibility Lamedb 2.x
- Fix import multiple files
- Fix convert menu and switcher
- Fix connection profiles bug
- Added Neutrino terrestrial services
- Bugfix

**Full Changelog**: https://github.com/ctlcltd/e2-sat-editor/compare/v1.6.0...v1.7.0

<hr /><em>This discussion was created from the release <a href='https://github.com/ctlcltd/e2-sat-editor/releases/tag/v1.7.0'>v1.7.0 release</a>.</em>

**Comment by p1j0** [comment]
Hello, I am trying to run the app on MacBook with M4, but it aways says that the file is damaged and cannot be started. I tried adding it as exception in the development tools, but without success. Am I missing something?
Thanks

**Reply by ctlcltd** [reply]
Thank you for the reporting!
I don't think you left anything out, unfortunately I don't have enough information for macOS under Silicon.
Currently the silicon version is **_testing_**. I have had only one negative feedback for Silicon, issue https://github.com/ctlcltd/e2-sat-editor/issues/25.
_Please consider:_ macOS may decide to quarantine files downloaded from the internet, in which case it may not be sufficient to allow the execution of not signed software. You may need to recursively remove the entire _Bundle.app_ package and its contents from the quarantine (extended file attribute: _com.apple.quarantine_).

The minimum requirement for current build on Silicon is macOS 14.0 (Sonoma).

_Below informations about the current situation regarding the builds and macOS._

_Premise:_ the software is tested under the Intel architecture. The software is not signed because I do not have a sufficient engagement to justify a cost of 99 USD per year for a free software (99 USD per year is the cost for an Apple Developer Program license).
I do not know if a security exception for a not signed software is sufficient to run a not signed software under the Silicon architecture.
A security exception can be granted on System Settings, Privacy & Security, Security, _Allow applications downloaded from ..._ This is the Apple documentation page about it: https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/

Changes have been made from software version 1.6 and 1.7 (_current_). The current workflow used to build the software for macOS Silicon is here: https://github.com/ctlcltd/e2-sat-editor/actions/workflows/build-macos-silicon.yml
Dependency binaries are now taken from MacPorts, Qt (for Community Edition) and libcurl4 respectively.
Binaries signing are stripped out, because the software is not signed.
Since software version 1.7 the QtCore.framework binary is patched to solve a mistake (`qtprfxpath` in the MacPorts and Homebrew _QtCore_ binary overtop the paths given to _dyld_ and it can loads libraries outside the given execution path). This binary works as expected under the Intel architecture.

**Reply by ctlcltd** [reply]
I did some tests, it runs correctly in a test environment on macOS 14 and macOS 15 under the Silicon arm64 architecture (screenshots attached). But, in a test environment quarantine and gatekeeper are limited. In this test it does not appear that the executable is corrupted.
_Please note_, all files downloaded from the internet are quarantined on macOS by default. If it does not run by removing it from quarantine and giving the exception for execution, I do not know what to say 😐
If I know more about it I will inform you in this discussion.

![Screenshot on Sonoma](https://github.com/user-attachments/assets/cf56949c-9b4a-4ce0-8b85-8d688eab0db4)
![Screenshot on Sequoia](https://github.com/user-attachments/assets/28c2f927-dae3-44e7-9215-d62e1fe96564)
